File: nodes/models/classification/creme/random_forest/random_forest.py
from utils.utils import myprint as print
from river import ensemble
from nodes.models.classification.creme.creme_model import CremeModel

class RandomForest(CremeModel):
    
    def __init__(self, *args):
        super().__init__(*args)
        max_features = self.node_config['maxFeatures']
        if max_features == "None":
            max_features = None
        if self.node_config['unlimitedDepth']:
            max_depth = None
        else:
            max_depth = self.node_config['maxDepth']
        self.set_model(
            ensemble.AdaptiveRandomForestClassifier(
                split_criterion='gini',
                max_depth=max_depth,
                max_features=max_features,
                n_models=self.node_config['numModels']
            )
        )

File: nodes/models/classification/creme/random_forest/random_forest.js
module.exports = function(RED) {
    function RandomForestNode(config) {

        const utils = require('../../../../../utils/utils');

        var node = this;

        node.topic = 'Random Forest'; // TODO: topic'leri düzenle
        // Corresponding python class
        node.pynode = 'RandomForest';

        node.hideProcessing = true;

        node.config = {
            propagateMode: parseInt(config.propagateAfter) < 2 ? "always" : config.propagateMode,
            propagateAfter: parseInt(config.propagateAfter),
            numModels: parseInt(config.numModels),
            unlimitedDepth: config.unlimitedDepth,
            maxDepth: parseInt(config.maxDepth)
        };

        if (config.loadModel)
            node.config.loadFrom = config.loadFolder.trim();

        if (config.maxFeaturesMethod == "value") {
            if (parseInt(config.maxFeaturesValue) == config.maxFeaturesValue) // if int
                node.config.maxFeatures = parseInt(config.maxFeaturesValue);
            else
                node.config.maxFeatures = parseFloat(config.maxFeaturesValue);
        } else {
            node.config.maxFeatures = config.maxFeaturesMethod;
        }

        utils.run(RED, node, config);

    }
    RED.nodes.registerType("random_forest", RandomForestNode);
}